### PR TITLE
Have the secure credentials in separate header files.

### PR DESCRIPTION
**Pull request by jboeuf**

### CODE REVIEW

**Comment by jboeuf** [comment]
looks like the python error on travis is unrelated....

File: src/cpp/client/secure_credentials.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
```
Comment (yang-g) [inline src/cpp/client/secure_credentials.h]: Header guard?

File: src/cpp/server/secure_server_credentials.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
```
Comment (yang-g) [inline src/cpp/server/secure_server_credentials.h]: same here.

**Comment by yang-g** [comment]
LGTM otherwise

**Comment by nicolasnoble** [comment]
It's not unusual for python to randomly fail on travis. You should have access to travis in order to restart jobs. Just log in using your github account.

File: src/cpp/client/secure_credentials.h (hunk 1 above)
Comment (jboeuf) [inline src/cpp/client/secure_credentials.h]: Good catch. Thanks. See new commit.

File: src/cpp/server/secure_server_credentials.h (hunk 2 above)
Comment (jboeuf) [inline src/cpp/server/secure_server_credentials.h]: ditto :)

**Comment by jboeuf** [comment]
Regarding travis, I addressed Yang's comments and that launched a new run of the tests. However, if those fail, I'm pretty sure that it's a python issue since I'm only touching the cpp code here that is not a python dependency.

**Comment by ctiller** [comment]
LGTM

**Pull request merged**